server/src/utils/callAutomationUtils.ts

```diff
@@ -0,0 +1,294 @@
+  callConnectionId: string,
+  options?: TranscriptionOptions
+): Promise<void> => {
+  console.log('Starting transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+
+  return await callConnection.getCallMedia().startTranscription(options);
+};
+
+export const stopTranscriptionForCall = async (callConnectionId: string): Promise<void> => {
+  console.log('Stopping transcription for call:', callConnectionId);
+  console.log(CALLCONNECTION_ID_TO_CORRELATION_ID);
+  const callConnection = await getCallAutomationClient().getCallConnection(callConnectionId);
+  return await callConnection.getCallMedia().stopTranscription();
+};
+
+/**
+ * We should be storing this based on the connectionId and not the correlationId
+ * We should re-write all the fetches based on the callId to use the serverCallId. This is
+ * because all of the clients that join can have different callId's and we need to make sure we are
+ * pulling the correct transcription data.
+ */
+export const TRANSCRIPTION_STORE: { [key: string]: Partial<CallTranscription> } = {};
+/**
+ * Used to map between the call connection id and the correlation id from both transcription and
+ * call automation events.
+ */
+export const CALLCONNECTION_ID_TO_CORRELATION_ID: {
+  [key: string]: { correlationId?: string; serverCallId: string };
+} = {};
+
+/**
+ * used to store the remote participants in the call
+ * This object is keyed off the callId and contains the communicationUserId and displayName of the remote participants
+ *
+ * Keeps track of all participants who have ever joined the call so we can show their display name in the transcription and summary.
+ */
+export const REMOTE_PARTICIPANTS_IN_CALL: {
+  [key: string]: { communicationUserId: string; displayName: string }[];
+} = {};
+
+/**
+ * used to store the local participant in the call
+ * This is keyed off the callId and contains the communicationUserId and displayName of the local participant
+ *
+ * This is important because the server can track the user across multiple calls if they change their display name so we can show
+ * the correct display name in the transcription and summary.
+ */
+export const LOCAL_PARTICIPANT: { [key: string]: { communicationUserId?: string; displayName?: string } } = {};
+
+export const getTranscriptionData = (serverCallId: string): CallTranscription | undefined => {
+  console.log('Getting transcription data for call:', serverCallId);
+  const connectionId = Object.keys(CALLCONNECTION_ID_TO_CORRELATION_ID).find((key) =>
+    CALLCONNECTION_ID_TO_CORRELATION_ID[key].serverCallId.includes(serverCallId)
+  );
+  if (!connectionId) {
+    return undefined;
+  }
+  const correlationId = CALLCONNECTION_ID_TO_CORRELATION_ID[connectionId]?.correlationId;
+  if (!correlationId) {
+    return undefined;
+  }
+  return TRANSCRIPTION_STORE[correlationId] as CallTranscription;
```

Comment:
How is this `CallTranscription` and not `Partial<CallTranscription>` now?

Reply:
the map can just be a CallTranscript now. I think there was originally some issues when setting up the metadata what the value would be for the transcription data itself and by making sure both are set when we start a new call and get that first event we can get rid of the partial